notifique/src/main/java/com/nathanrassi/notifique/NotifiqueListenerService.kt

```diff
@@ -1,20 +1,65 @@
 package com.nathanrassi.notifique
 
+import android.app.Notification.EXTRA_TEXT
+import android.app.Notification.EXTRA_TITLE
 import android.service.notification.NotificationListenerService
 import android.service.notification.StatusBarNotification
 import dagger.android.AndroidInjection
 import javax.inject.Inject
 
 class NotifiqueListenerService : NotificationListenerService() {
   @Inject internal lateinit var dao: Notifique.Dao
+  private val packageNameToIds = mutableMapOf<String, MutableList<Int>>()
 
   override fun onCreate() {
     AndroidInjection.inject(this)
   }
 
   override fun onNotificationPosted(sbn: StatusBarNotification) {
+    val packageName = sbn.packageName
+    val notificationId = sbn.id
+    if (storeIfUnique(packageName, notificationId)) {
+      val notificationExtras = sbn.notification.extras
+      val message = notificationExtras.getString(EXTRA_TEXT).toString()
+      val title = notificationExtras.getString(EXTRA_TITLE).toString()
+      if (!message.isNullOrEmpty() && !title.isNullOrEmpty()) {
```

Comment:
no, toString() never returns null. in fact, the IDE is probably smart enough to warn you with that code.
remove all the toString calls except in the constructor.